### ISSUE TITLE
Teleport SSO with OIDC auth connector

### ISSUE THREAD

**Issue by programmerq** [gravitational/teleport#9736]
### What
What would you like Teleport to do differently?

Teleport should handle a 401 from the userinfo endpoint when working with an oidc provider.

### How
How would you implement this?

### Why
Why do you need this?

Security policy with a company internal oidc provider prohibits access to that endpoint and teleport errors out when it encounters the 401.

```
2022-01-10T23:56:35Z DEBU [AUTH]      Fetching OIDC claims from UserInfo endpoint: "https://id.example.com/adfs/userinfo". auth/oidc.go:590
2022-01-10T23:56:35Z DEBU [AUTH]      Unable to fetch UserInfo claims: bad status code: 401. auth/oidc.go:801
2022-01-10T23:56:35Z INFO [AUDIT]     user.login cluster_name:teleport code:T1001W ei:0 error:[bad status code: 401] event:user.login message:"unable to fetch UserInfo claims\n\tbad status code: 401" method:oidc success:false time:2022-01-10T23:56:35.899Z uid:aaabbb11-2233-4444-0000-778891827364 events/emitter.go:324
```

### Workaround
If a workaround exists, please include it.

none gz#3733

**Comment by espadolini** [comment]
Technically there could be rules that deny based on OIDC claims; there's precedent for it at least, the current version of the OIDC connector will stop fetching group claims from google workspace after 100 pages' worth, but it's still something to think about.

Perhaps an `allow_userinfo_failure` option in the connector?

**Comment by zmb3** [comment]
Is this allowed per the OIDC spec? I'm not crazy about adding special cases for non-compliant OIDC implementations.

**Comment by espadolini** [comment]
Looks like it's allowed, #9768

**Comment by programmerq** [comment]
This was fixed by #9951
